js/net_test.ts

```diff
@@ -1,13 +1,34 @@
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
-import { testPerm, assert, assertEquals } from "./test_util.ts";
+import { testPerm, assert, assertEquals, runIfMain } from "./test_util.ts";
+
+function deferred(): {
+  resolve: () => void;
+  reject: () => void;
+  promise: Promise<void>;
+} {
+  let resolve: () => void;
+  let reject: () => void;
+  const promise = new Promise<void>(
+    (a, b): void => {
+      resolve = a;
+      reject = b;
+    }
+  );
+  return { resolve, reject, promise };
+}
+
+let port = 4500;
+function createAddr(): string {
```

Comment:
Not really a fan of this part because using this we are not sure the socket is closed.
Adding `await new Promise(resolve => setTimeout(resolve, 1000));` fixes the issue.
After discussing about it with @ry , the problem is not about the `close` operation of the listenner but the time the network takes to free the tcp socket.

Reply:
> using this we are not sure the socket is closed.

What do you mean? I think using different port with each test won't affect test results.

Reply:
No it don't affect the test at all. Just noticed the problem of port already used. By using different ports you're not ensuring the listenner is closed and has liberated the port.

Reply:
Sure. But there are a gap between expected behaviour and actual one. Test codes are written with expected behaviour about listener but results are broken. 

>you're not ensuring the listenner is closed

Correct. Tough there is no sufficient evidence, I doubt that `listener.close()` won't reject pending accept synchronously and release listening port immediately...

This is definitely a cause of flaky CI on each platform. I think `listener.listen()` and `listen.close()` should be asynchronous for ensuring listening/closing port on developer side. @ry how do you think?